django/librarian/utils/process_engine.py

```diff
@@ -459,3 +481,47 @@ def _pdf_to_html_using_azure(content):
         html += page_end_tag
 
     return html
+
+
+def pdf_to_text_azure_read(content: bytes) -> str:
+
+    from azure.ai.formrecognizer import DocumentAnalysisClient
+    from azure.core.credentials import AzureKeyCredential
+
+    document_analysis_client = DocumentAnalysisClient(
+        endpoint=settings.AZURE_COGNITIVE_SERVICE_ENDPOINT,
+        credential=AzureKeyCredential(settings.AZURE_COGNITIVE_SERVICE_KEY),
+    )
+
+    poller = document_analysis_client.begin_analyze_document("prebuilt-read", content)
+    result = poller.result()
+
+    num_pages = len(result.pages)
+    cost = Cost.objects.new(cost_type="doc-ai-read", count=num_pages)
+
+    p_chunks = []
+    for page in result.pages:
+        for line in page.lines:
+            chunk = {
+                "page_number": page.page_number,
+                "text": line.content + "\n",
+            }
+            p_chunks.append(chunk)
+
+    text = ""
+    cur_page = None
+    for _, chunk in enumerate(p_chunks, 1):
```

Comment:
I'm *not* sorting these by X,Y coordinates. I found that actually made it worse.